gobblin-runtime/src/main/java/gobblin/scheduler/JobScheduler.java

```diff
@@ -404,46 +402,44 @@ private void scheduleGeneralConfiguredJobs()
    *   is called on the changes is not controlled by Gobblin, but instead by the monitor itself.
    * </p>
    */
-  private void startJobConfigFileMonitor()
+  private void startGeneralJobConfigFileMonitor()
       throws Exception {
-    LOG.info("Monitor debugging: " + this.properties.getProperty(ConfigurationKeys.JOB_CONFIG_FILE_DIR_KEY));
-    final File jobConfigFileDir = new File(this.properties.getProperty(ConfigurationKeys.JOB_CONFIG_FILE_DIR_KEY));
-    FileAlterationListener listener = new FileAlterationListenerAdaptor() {
-      /**
-       * Called when a new job configuration file is dropped in.
-       */
+    final Path jobConfigFileDirPath =
+        new Path(this.properties.getProperty(ConfigurationKeys.JOB_CONFIG_FILE_GENERAL_PATH_KEY));
+    PathAlterationListener listener = new PathAlterationListenerAdaptor() {
       @Override
-      public void onFileCreate(File file) {
-        String fileExtension = Files.getFileExtension(file.getName());
-        if (!JobScheduler.this.jobConfigFileExtensions.contains(fileExtension)) {
-          // Not a job configuration file, ignore.
+      public void onFileCreate(Path path) {
+        String fileExntensionFromPath = path.getName().substring(path.getName().lastIndexOf('.') + 1);
```

Comment:
There is a lot of code repetition between here and `onFileChange`, can they be refactored a bit?